synapse/storage/util/id_generators.py

```diff
@@ -425,7 +456,7 @@ def _add_persisted_position(self, new_id: int):
         # We move the current min position up if the minimum current positions
         # of all instances is higher (since by definition all positions less
         # that that have been persisted).
-        min_curr = min(self._current_positions.values())
+        min_curr = min(self._current_positions.values(), default=0)
```

Comment:
I'm struggling to see how this is directly related to this PR? The change looks fine though.

Reply:
Sigh, sorry, that is a somewhat unrelated fix that broke that particular sytest. Basically, if the first time a worker see's an ID go past it is out of order, you can get here without anything having been added to current positions, causing this to explode.